🐛 [Bug] Cannot convert simple torchscript containing two torch.nn.Upsample operations

##  Bug Description

Scripting a simple "network" containing two `torch.nn.Upsample` modules and trying to convert the resulting torchscript does not work.

## To Reproduce

Steps to reproduce the behavior:

1. Generate a torchscript of the Network below, with `torch.jit.script`.
2. Try to convert to TensorRT with torch_tensorrt (I tried both in C++, on linux and in python)

```python
class Network(torch.nn.Module):
    def __init__(self):
        super(Network, self).__init__()
        # NOTE:
        # * Specifying a single float as scale_factor or the 2d tuple doesn't change the behavior
        self.upsample1 = torch.nn.Upsample(
            scale_factor=(2.0, 2.0), mode="bilinear", align_corners=False
        )
        self.upsample2 = torch.nn.Upsample(
            scale_factor=(2.0, 2.0), mode="bilinear", align_corners=False
        )

    def forward(self, X: torch.Tensor) -> torch.Tensor:
        # NOTE:
        # * Using the same self.upsample() module doesn't work as well
        # * Computing and returning out doesn't work as well
        out1 = self.upsample1(X)
        out2 = self.upsample2(X)
        # out = out1 + out2
        return out1
```

## Expected behavior

The conversions succeeds and a new valid torchscript is obtained.

## Environment

I managed to reproduce the bug both when using pytorch 1.11 and torch-tensorRT 1.1.0 and using pytorch 1.13.1 and torch-tensorRT main.

### Torch-TensorRT 1.1.0

 - Torch-TensorRT Version (e.g. 1.0.0): 1.1.0
 - PyTorch Version (e.g. 1.0): 1.11
 - CPU Architecture: x86_64
 - OS (e.g., Linux): Linux, Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip for the python package used to generate the torchscript, source for the C++ dependency linked to Torch-TensorRT
 - Build command you used (if compiling from source): Compiled with CMake
 - Python version: 3.8.16
 - CUDA version: 11.3
 - GPU models and configuration: NVIDIA GeForce RTX 2070 SUPER, SM Capability: 7.5
 - Any other relevant information:

When using torch-tensorRT 1.1.0, I get the following error:
```
DEBUG: [Torch-TensorRT] - Registering input/output torch::jit::Value for segmented graphs
terminate called after throwing an instance of 'c10::Error'
  what():  Expected Tensor but got Uninitialized


Exception raised from reportToTensorTypeError at /home/ubuntu/buildAgent/temp/buildTmp/conan_home/.conan/data/libtorch/1.11.0-5/cognex/stable/build/b8fcba865a68ae644e8a8bfa868ef00d13dc8a17/source_subfolder/aten/src/ATen/core/ivalue.cpp:908 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x7f66313e708b in /mnt/caches/conan/data/libtorch/1.11.0-5/cognex/stable/package/b8fcba865a68ae644e8a8bfa868ef00d13dc8a17/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xce (0x7f66313e2a5e in /mnt/caches/conan/data/libtorch/1.11.0-5/cognex/stable/package/b8fcba865a68ae644e8a8bfa868ef00d13dc8a17/lib/libc10.so)
frame #2: c10::IValue::reportToTensorTypeError() const + 0x64 (0x7f6634480064 in /mnt/caches/conan/data/libtorch/1.11.0-5/cognex/stable/package/b8fcba865a68ae644e8a8bfa868ef00d13dc8a17/lib/libtorch_cpu.so)
frame #3: torch_tensorrt::core::partitioning::getSegmentsOutputByRunning(torch_tensorrt::core::partitioning::SegmentedBlock&, std::unordered_map<torch::jit::Value const*, c10::IValue, std::hash<torch::jit::Value const*>, std::equal_to<torch::jit::Value const*>, std::allocator<std::pair<torch::jit::Value const* const, c10::IValue> > >&, torch_tensorrt::core::partitioning::PartitionInfo const&) + 0x15a7 (0x7f6639fd8797 in /mnt/caches/conan/data/torch-tensorrt/1.1.0-6/cognex/stable/package/c2bc9df050867ddb44d73bc998d3c8ceef4f763a/lib/libtorchtrt.so)
frame #4: torch_tensorrt::core::partitioning::runShapeAnalysis(std::vector<torch_tensorrt::core::partitioning::SegmentedBlock, std::allocator<torch_tensorrt::core::partitioning::SegmentedBlock> >&, std::unordered_map<torch::jit::Value const*, c10::IValue, std::hash<torch::jit::Value const*>, std::equal_to<torch::jit::Value const*>, std::allocator<std::pair<torch::jit::Value const* const, c10::IValue> > >&, torch_tensorrt::core::partitioning::PartitionInfo const&) + 0x81 (0x7f6639fda581 in /mnt/caches/conan/data/torch-tensorrt/1.1.0-6/cognex/stable/package/c2bc9df050867ddb44d73bc998d3c8ceef4f763a/lib/libtorchtrt.so)
frame #5: torch_tensorrt::core::partitioning::Partition(torch::jit::Block*, std::unordered_map<torch::jit::Value const*, c10::IValue, std::hash<torch::jit::Value const*>, std::equal_to<torch::jit::Value const*>, std::allocator<std::pair<torch::jit::Value const* const, c10::IValue> > >&, torch_tensorrt::core::partitioning::PartitionInfo const&) + 0x19b (0x7f6639fe587b in /mnt/caches/conan/data/torch-tensorrt/1.1.0-6/cognex/stable/package/c2bc9df050867ddb44d73bc998d3c8ceef4f763a/lib/libtorchtrt.so)
frame #6: torch_tensorrt::core::ConstructFallbackGraph(torch::jit::Module&, torch::jit::Block*, std::unordered_map<torch::jit::Value const*, c10::IValue, std::hash<torch::jit::Value const*>, std::equal_to<torch::jit::Value const*>, std::allocator<std::pair<torch::jit::Value const* const, c10::IValue> > >, torch_tensorrt::core::CompileSpec, std::map<torch::jit::Value*, c10::IValue, std::less<torch::jit::Value*>, std::allocator<std::pair<torch::jit::Value* const, c10::IValue> > >) + 0xff (0x7f663a00006f in /mnt/caches/conan/data/torch-tensorrt/1.1.0-6/cognex/stable/package/c2bc9df050867ddb44d73bc998d3c8ceef4f763a/lib/libtorchtrt.so)
frame #7: torch_tensorrt::core::CompileGraph(torch::jit::Module const&, torch_tensorrt::core::CompileSpec) + 0x980 (0x7f663a002a00 in /mnt/caches/conan/data/torch-tensorrt/1.1.0-6/cognex/stable/package/c2bc9df050867ddb44d73bc998d3c8ceef4f763a/lib/libtorchtrt.so)
frame #8: torch_tensorrt::torchscript::compile(torch::jit::Module const&, torch_tensorrt::torchscript::CompileSpec) + 0x5b7 (0x7f6639e9e7f7 in /mnt/caches/conan/data/torch-tensorrt/1.1.0-6/cognex/stable/package/c2bc9df050867ddb44d73bc998d3c8ceef4f763a/lib/libtorchtrt.so)
frame #9: <unknown function> + 0x5ec8 (0x56183fd18ec8 in ./build/src/interpolate_tensorrt)
frame #10: __libc_start_main + 0xf3 (0x7f65fe02f083 in /lib/x86_64-linux-gnu/libc.so.6)
frame #11: <unknown function> + 0x58ee (0x56183fd188ee in ./build/src/interpolate_tensorrt)
```

That looked kind of similar to [this issue](https://github.com/pytorch/TensorRT/issues/1282) and patching Torch-TensorRT with [this PR](https://github.com/pytorch/TensorRT/pull/1345) makes the behavior exactly the same as in the second case (i.e. when using pytorch 1.13.1 and torch-tensorRT main).

### Torch-TensorRT main (commit 861edd03a510c600146575836b02c993ac386b00)

 - Torch-TensorRT Version (e.g. 1.0.0): main, commit 861edd03a510c600146575836b02c993ac386b00
 - PyTorch Version (e.g. 1.0): 1.13.1
 - CPU Architecture: x86_64
 - OS (e.g., Linux): Linux, Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip for the python package used to generate the torchscript, source for the C++ dependency linked to Torch-TensorRT
 - Build command you used (if compiling from source): Compiled with CMake
 - Python version: 3.8.16
 - CUDA version: 11.8
 - GPU models and configuration: NVIDIA GeForce RTX 2070 SUPER, SM Capability: 7.5
 - Any other relevant information:

When using torch-TensorRT main, the conversion just hangs for ever after 
```
GRAPH: [Torch-TensorRT] - Torch-TensorRT.TorchScript Graph Lowering
```


## Additional context

Interestingly, when using the tracing mechanism of pytorch to generate the torchscript, everything _seems_ fine (I didn't check the results, but the conversion finishes properly).
Also, when scripting with pytorch 1.9, everything works fine 🤯 

The thing I noticed is that pytorch changed slightly the `torch.nn.interpolate` API and I am wondering if that could explain (at least partially) the problem:
- Before pytorch 1.11: `torch.nn.functional.interpolate(input, size=None, scale_factor=None, mode='nearest', align_corners=None, recompute_scale_factor=None)`
- PyTorch 1.11 and more recent: `torch.nn.functional.interpolate(input, size=None, scale_factor=None, mode='nearest', align_corners=None, recompute_scale_factor=None, antialias=False)`

See the attached .zip file containing a python file to generate the torchscript.
[upsample.zip](https://github.com/pytorch/TensorRT/files/11234555/upsample.zip)

Let me know if you need more details to reproduce the problem. Thanks!

Thanks @gs-olive for having taken a look at this issue so quickly! Nice that you could reproduce it!
Following up on that, here is another simple Network (though slightly less trivial than the one above). 

<details>
  <summary>Network definition</summary>

```python
import torch
import torch.nn as nn
import torch.nn.functional as F
import argparse


class Block(nn.Module):
    def __init__(self, in_channel, out_channel):
        super(Block, self).__init__()
        self.conv = nn.Conv2d(
            in_channel, out_channel, kernel_size=3, stride=1, padding=1, bias=False
        )
        self.norm = nn.BatchNorm2d(out_channel)
        self.relu = nn.ReLU(inplace=True)
        self.maxpool = nn.MaxPool2d(
            kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False
        )

    def forward(self, x):
        out = self.conv(x)
        out = self.norm(out)
        out = self.relu(out)
        out = self.maxpool(out)
        return out


class Network(torch.nn.Module):
    def __init__(self, num_classes=2):
        super(Network, self).__init__()
        self.num_classes = num_classes

        self.block1 = Block(3, 32)
        self.block2 = Block(32, 64)

        self.upsample1 = nn.Upsample(
            scale_factor=2, mode="bilinear", align_corners=False
        )

        self.upsample2 = nn.Upsample(
            scale_factor=2, mode="bilinear", align_corners=False
        )

        self.conv = nn.Conv2d(64, num_classes, 1, bias=True)

    def forward(self, x):
        out = self.block1(x)
        out = self.block2(out)

        gclayer1 = self.upsample1(out)
        gclayer2 = self.upsample2(gclayer1)

        out = self.conv(gclayer2)
        return out

```

</details>

Interestingly, when converting a torchscript generated from that network using `torch.jit.script` on linux, the behaviour is the same as with the trivial network previously shared: using torch-tensorrt [recent commit from main](https://github.com/pytorch/TensorRT/commit/861edd03a510c600146575836b02c993ac386b00), it hangs (i.e. the operation `torch::jit::EliminateExceptions` does not complete). 
Nevertheless, when converting the torchscript **on windows** using the same torch-tensorrt [recent commit from main](https://github.com/pytorch/TensorRT/commit/861edd03a510c600146575836b02c993ac386b00), **it works**! Note that it works both when the torchscript is generated on windows or on linux.

Both graphs, as printed by Torch-TensorRT when calling `compile` are included in [archive.zip](https://github.com/pytorch/TensorRT/files/11272551/archive.zip), as well as the mini_net.py network definition/torchscript generation file. In summary, here is a diff between the graphs, as printed by Torch-TensorRT:

<details>
<summary>Graph diff</summary>

```
--- graph_lin.py	2023-04-19 13:28:59.000000000 +0200
+++ graph_win.py	2023-04-19 13:28:52.000000000 +0200
@@ -1,16 +1,13 @@
 GRAPH: [Torch-TensorRT] - After freeze: graph(%self.1 : __torch__.___torch_mangle_0.Network,
       %x.1 : Tensor):
-  %224 : str[] = prim::Constant[value=["nearest", "area", "nearest-exact"]]()
   %115 : str = prim::Constant[value="bilinear"]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/modules/upsampling.py:153:66
   %114 : float = prim::Constant[value=2.]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/modules/upsampling.py:153:47
   %113 : int = prim::Constant[value=5]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3892:22
   %112 : int = prim::Constant[value=3]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3888:22
   %111 : int = prim::Constant[value=4]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3885:76
-  %109 : str = prim::Constant[value="align_corners option can only be set with the interpolating modes: linear | bilinear | bicubic | trilinear"]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3815:16
   %108 : str = prim::Constant[value="Got 3D input, but bilinear mode needs 4D input"]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3930:34
   %107 : str = prim::Constant[value="Got 5D input, but bilinear mode needs 4D input"]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3940:34
   %106 : str = prim::Constant[value="Input Error: Only 3D, 4D and 5D input Tensors supported (got {}D) for the modes: nearest | linear | bilinear | bicubic | trilinear | area | nearest-exact (got {})"]() # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3943:8
-  %101 : str = prim::Constant[value="AssertionError: "]() # :0:0
   %223 : int[] = prim::Constant[value=[0, 0]]()
   %222 : int[] = prim::Constant[value=[2, 2]]()
   %221 : int[] = prim::Constant[value=[1, 1]]()
@@ -38,15 +35,6 @@
   %out1.1 : Tensor = aten::relu_(%out0.1) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:1440:17
   %out0.2 : Tensor = aten::max_pool2d(%out1.1, %222, %222, %223, %221, %self.block1.norm.training) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:797:11
   %117 : Tensor = prim::Uninitialized() # :0:0
-  %118 : bool? = prim::Uninitialized() # :0:0
-  %119 : bool = prim::Uninitialized() # :0:0
-  %121 : bool = aten::__contains__(%224, %115) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3812:7
-  %align_corners0.1 : bool? = prim::If(%121) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3812:4
-    block0():
-       = prim::RaiseException(%109, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3814:12
-      -> (%118)
-    block1():
-      -> (%self.block1.norm.training)
   %123 : int = aten::dim(%out0.2) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3822:10
   %dim.2 : int = aten::sub(%123, %18) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3822:10
   %scale_factors2.2 : float[] = prim::ListConstruct()
@@ -57,15 +45,7 @@
   %129 : bool = aten::eq(%123, %111) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3915:7
   %gclayer1.1 : Tensor = prim::If(%129) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3915:4
     block0():
-      %132 : bool = aten::__isnot__(%align_corners0.1, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3916:15
-      %align_corners6.1 : bool = prim::If(%132) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3916:8
-        block0():
-          %align_corners7.2 : bool = prim::unchecked_cast(%align_corners0.1)
-          -> (%align_corners7.2)
-        block1():
-           = prim::RaiseException(%101, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3916:8
-          -> (%119)
-      %135 : Tensor = aten::upsample_bilinear2d(%out0.2, %self.block1.conv.bias, %align_corners6.1, %scale_factors2.2) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3919:15
+      %135 : Tensor = aten::upsample_bilinear2d(%out0.2, %self.block1.conv.bias, %self.block1.norm.training, %scale_factors2.2) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3919:15
       -> (%135)
     block1():
       %137 : bool = aten::eq(%123, %112) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3929:7
@@ -85,12 +65,6 @@
       %143 : str = aten::format(%106, %123, %115) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3943:8
        = prim::RaiseException(%143, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3942:4
       -> (%117)
-  %align_corners0 : bool? = prim::If(%121) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3812:4
-    block0():
-       = prim::RaiseException(%109, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3814:12
-      -> (%118)
-    block1():
-      -> (%self.block1.norm.training)
   %167 : int = aten::dim(%gclayer1.1) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3822:10
   %dim.1 : int = aten::sub(%167, %18) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3822:10
   %scale_factors2.1 : float[] = prim::ListConstruct()
@@ -101,15 +75,7 @@
   %173 : bool = aten::eq(%167, %111) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3915:7
   %gclayer2.1 : Tensor = prim::If(%173) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3915:4
     block0():
-      %176 : bool = aten::__isnot__(%align_corners0, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3916:15
-      %align_corners6 : bool = prim::If(%176) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3916:8
-        block0():
-          %align_corners7.1 : bool = prim::unchecked_cast(%align_corners0)
-          -> (%align_corners7.1)
-        block1():
-           = prim::RaiseException(%101, %self.block1.conv.bias) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3916:8
-          -> (%119)
-      %179 : Tensor = aten::upsample_bilinear2d(%gclayer1.1, %self.block1.conv.bias, %align_corners6, %scale_factors2.1) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3919:15
+      %179 : Tensor = aten::upsample_bilinear2d(%gclayer1.1, %self.block1.conv.bias, %self.block1.norm.training, %scale_factors2.1) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3919:15
       -> (%179)
     block1():
       %181 : bool = aten::eq(%167, %112) # /home/gcuendet/.pyenv/versions/pytorch1.11/lib/python3.8/site-packages/torch/nn/functional.py:3929:7

```
</details>

Please let me know if I can provide more details to help solve this issue!

Just for completeness: using `Upsample` directly in the network definition still does not seem to work, even with the changes proposed above and [the fix in `torch::jit::EliminateException`](https://github.com/pytorch/TensorRT/pull/1859).
I am now observing the following error, in `shape_analysis.cpp` with the Network described in [this comment](https://github.com/pytorch/TensorRT/issues/1823#issuecomment-1514586730):

```
GRAPH: [Torch-TensorRT] - Running shape analysis on block Segment Block @1:
    Target: Torch

    Graph: graph(%1 : bool,
      %4 : bool?):
  %self.block1.norm.training.15 : bool = prim::Constant[value=0]()
  %3 : str = prim::Constant[value="builtins.ValueError"]()
  %2 : str = prim::Constant[value="align_corners option can only be set with the interpolating modes: linear | bilinear | bicubic | trilinear"]()
  %align_corners0.1 : bool? = prim::If(%1)
    block0():
       = prim::RaiseException(%2, %3)
      -> (%4)
    block1():
      -> (%self.block1.norm.training.15)
  return (%align_corners0.1)


terminate called after throwing an instance of 'torch_tensorrt::Error'
  what():  [Error thrown at core/partitioning/shape_analysis.cpp:187] Expected ivalues_maps.count(input) to be true but got false
Could not find torch::jit::Value* 474 produced from %474 : bool? = prim::Uninitialized() # :0:0 in lowering graph for mini graph input.
```

This seems linked to the check on the validity of the `align_corners` option (checking that the interpolating mode is one of linear, bilinear, bicubic or trilinear) and the fact that, when it's invalid, a `bool? = prim::Uninitialized()` is returned, which seems to cause the shape analysis to fail..? This is not simplified by the exception elimination pass, since the block actually returns something.

A workaround for this is to not call `Upsample`, but rather directly `upsample_bilinear2d` (or one of the other functions corresponding to the correct interpolating mode and dimensions). 

I also include the lowered graph below (lowered with all the changes described above):
<details>
<summary>Lowered graph</summary>

```python
INFO: [Torch-TensorRT] - Lowered Graph: graph(%x.1 : Tensor):                                                                                                                                                                                                                                                                                                                                                                                                                                  [540/18335]
  %4 : float = prim::Constant[value=2.]()
  %14 : int[] = prim::Constant[value=[0, 0]]()
  %16 : int[] = prim::Constant[value=[1, 1]]()
  %17 : int = prim::Constant[value=1]()
  %18 : bool = prim::Constant[value=1]()
  %self.conv.bias : Float(2, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=0.01 *  7.4246 -8.4862 [ CUDAFloatType{2} ]]()
  %self.conv.weight : Float(2, 64, 1, 1, strides=[64, 1, 1, 1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.conv.weight : Float(32, 3, 3, 3, strides=[27, 9, 3, 1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.conv.bias.15 : NoneType = prim::Constant()
  %self.block1.norm.running_var : Float(32, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.norm.running_mean : Float(32, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.norm.training.15 : bool = prim::Constant[value=0]()
  %452 : float = prim::Constant[value=0.10000000000000001]()
  %453 : float = prim::Constant[value=1.0000000000000001e-05]()
  %455 : int[] = prim::Constant[value=[2, 2]]()
  %self.block2.conv.weight : Float(64, 32, 3, 3, strides=[288, 9, 3, 1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block2.norm.running_var : Float(64, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block2.norm.running_mean : Float(64, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %460 : str[] = prim::Constant[value=["nearest", "area", "nearest-exact"]]()
  %461 : str = prim::Constant[value="bilinear"]()
  %462 : str = prim::Constant[value="builtins.ValueError"]()
  %463 : str = prim::Constant[value="align_corners option can only be set with the interpolating modes: linear | bilinear | bicubic | trilinear"]()
  %464 : int = prim::Constant[value=2]()
  %579 : bool = prim::Constant[value=0]()
  %580 : int[] = prim::Constant[value=[0, 0]]()
  %581 : Tensor = aten::_convolution(%x.1, %self.block1.conv.weight, %self.block1.conv.bias.15, %16, %16, %16, %579, %580, %17, %579, %579, %579, %579)
  %out1.2 : Tensor = aten::batch_norm(%581, %self.block1.norm.running_var, %self.block1.norm.running_mean, %self.block1.norm.running_mean, %self.block1.norm.running_var, %self.block1.norm.training.15, %452, %453, %18)
  %467 : Tensor = aten::relu(%out1.2)
  %out.2 : Tensor = aten::max_pool2d(%467, %455, %455, %14, %16, %self.block1.norm.training.15)
  %582 : bool = prim::Constant[value=0]()
  %583 : int[] = prim::Constant[value=[0, 0]]()
  %584 : Tensor = aten::_convolution(%out.2, %self.block2.conv.weight, %self.block1.conv.bias.15, %16, %16, %16, %582, %583, %17, %582, %582, %582, %582)
  %out0.1 : Tensor = aten::batch_norm(%584, %self.block2.norm.running_var, %self.block2.norm.running_mean, %self.block2.norm.running_mean, %self.block2.norm.running_var, %self.block1.norm.training.15, %452, %453, %18)
  %471 : Tensor = aten::relu(%out0.1)
  %out0.2 : Tensor = aten::max_pool2d(%471, %455, %455, %14, %16, %self.block1.norm.training.15)
  %474 : bool? = prim::Uninitialized() # :0:0
  %475 : bool = prim::Uninitialized() # :0:0
  %476 : bool = aten::__contains__(%460, %461)
  %align_corners0.1 : bool? = prim::If(%476)
    block0():
       = prim::RaiseException(%463, %462)
      -> (%474)
    block1():
      -> (%self.block1.norm.training.15)
  %478 : int = aten::dim(%out0.2)
  %dim.2 : int = aten::sub(%478, %464)
  %scale_factors2.2 : float[] = prim::ListConstruct()
   = prim::Loop(%dim.2, %18)
    block0(%49 : int):
      %50 : float[] = aten::append(%scale_factors2.2, %4)
      -> (%18)
  %480 : int = prim::Constant[value=4]()
  %481 : str = prim::Constant[value="Input Error: Only 3D, 4D and 5D input Tensors supported (got {}D) for the modes: nearest | linear | bilinear | bicubic | trilinear | area | nearest-exact (got {})"]()
  %482 : str = prim::Constant[value="AssertionError: "]()
  %485 : str = prim::Constant[value="Got 3D input, but bilinear mode needs 4D input"]()
  %486 : int = prim::Constant[value=3]()
  %487 : str = prim::Constant[value="builtins.NotImplementedError"]()
  %488 : int = prim::Constant[value=5]()
  %489 : str = prim::Constant[value="Got 5D input, but bilinear mode needs 4D input"]()
  %494 : bool = aten::eq(%478, %480)
  %571 : bool = aten::__isnot__(%align_corners0.1, %self.block1.conv.bias.15)
  %align_corners6.2 : bool = prim::If(%571)
    block0():
      %align_corners7.3 : bool = prim::unchecked_cast(%align_corners0.1)
      -> (%align_corners7.3)
    block1():
       = prim::RaiseException(%482, %self.block1.conv.bias.15)
      -> (%475)
  %574 : Tensor = aten::upsample_bilinear2d(%out0.2, %self.block1.conv.bias.15, %align_corners6.2, %scale_factors2.2)
   = prim::If(%494)
    block0():
       = prim::If(%571)
        block0():
          -> ()
        block1():
           = prim::RaiseException(%482, %self.block1.conv.bias.15)
          -> ()
      -> ()
    block1():
      %500 : bool = aten::eq(%478, %486)
       = prim::If(%500)
        block0():
           = prim::RaiseException(%485, %487)
          -> ()
        block1():
          -> ()
      %501 : bool = aten::eq(%478, %488)
       = prim::If(%501)
        block0():
           = prim::RaiseException(%489, %487)
          -> ()
        block1():
          -> ()
      %502 : str = aten::format(%481, %478, %461)
       = prim::RaiseException(%502, %487)
      -> ()
  %align_corners0 : bool? = prim::If(%self.block1.norm.training.15)
    block0():
       = prim::RaiseException(%463, %462)
      -> (%474)
    block1():
      -> (%self.block1.norm.training.15)
  %504 : int = aten::dim(%574)
  %dim.1 : int = aten::sub(%504, %464)
  %scale_factors2.1 : float[] = prim::ListConstruct()
   = prim::Loop(%dim.1, %18)
    block0(%64 : int):
      %65 : float[] = aten::append(%scale_factors2.1, %4)
      -> (%18)
  %516 : bool = aten::eq(%504, %480)
  %575 : bool = aten::__isnot__(%align_corners0, %self.block1.conv.bias.15)
  %align_corners6.6 : bool = prim::If(%575)
    block0():
      %align_corners7.7 : bool = prim::unchecked_cast(%align_corners0)
      -> (%align_corners7.7)
    block1():
       = prim::RaiseException(%482, %self.block1.conv.bias.15)
      -> (%475)
  %578 : Tensor = aten::upsample_bilinear2d(%574, %self.block1.conv.bias.15, %align_corners6.6, %scale_factors2.1)
   = prim::If(%516)
    block0():
       = prim::If(%575)
        block0():
          -> ()
        block1():
           = prim::RaiseException(%482, %self.block1.conv.bias.15)
          -> ()
      -> ()
    block1():
      %522 : bool = aten::eq(%504, %486)
       = prim::If(%522)
        block0():
           = prim::RaiseException(%485, %487)
          -> ()
        block1():
          -> ()
      %523 : bool = aten::eq(%504, %488)
       = prim::If(%523)
        block0():
           = prim::RaiseException(%489, %487)
          -> ()
        block1():
          -> ()
      %524 : str = aten::format(%481, %504, %461)
       = prim::RaiseException(%524, %487)
      -> ()
  %585 : bool = prim::Constant[value=0]()
  %586 : int[] = prim::Constant[value=[0, 0]]()
  %587 : Tensor = aten::_convolution(%578, %self.conv.weight, %self.conv.bias, %16, %14, %16, %585, %586, %17, %585, %585, %585, %585)
  return (%587)
```

</details>

Thanks @bowang007 . Did you test on the small network [described above](https://github.com/pytorch/TensorRT/issues/1823#issuecomment-1514586730)? Is it working for you?
The conversion is not working for me. I get the following error:

```
GRAPH: [Torch-TensorRT] - Running shape analysis on block Segment Block @3:
    Target: Torch

    Graph: graph(%out0.2 : Tensor,
      %align_corners0.1 : bool?):
  %self.block1.conv.bias.15 : NoneType = prim::Constant()
  %10 : int = prim::Constant[value=4]()
  %8 : float = prim::Constant[value=2.]()
  %5 : bool = prim::Constant[value=1]()
  %3 : int = prim::Constant[value=2]()
  %0 : int = aten::dim(%out0.2)
  %dim.2 : int = aten::sub(%0, %3)
  %scale_factors2.2 : float[] = prim::ListConstruct()
   = prim::Loop(%dim.2, %5)
    block0(%6 : int):
      %7 : float[] = aten::append(%scale_factors2.2, %8)
      -> (%5)
  %9 : bool = aten::eq(%0, %10)
  %11 : bool = aten::__isnot__(%align_corners0.1, %self.block1.conv.bias.15)
  return (%scale_factors2.2, %0, %9, %11)


terminate called after throwing an instance of 'torch_tensorrt::Error'
  what():  [Error thrown at core/partitioning/shape_analysis.cpp:212] Expected to find type bool? for value align_corners0.1 but get nothing.
```

For reference the lowered graph is the following.
<details>
<summary>Lowered graph</summary>

```python
INFO: [Torch-TensorRT] - Lowered Graph: graph(%x.1 : Tensor):                                                                                                                                                                                                                                                                                                                                                                                                                                  [588/19311]
  %4 : float = prim::Constant[value=2.]()
  %14 : int[] = prim::Constant[value=[0, 0]]()
  %16 : int[] = prim::Constant[value=[1, 1]]()
  %17 : int = prim::Constant[value=1]()
  %18 : bool = prim::Constant[value=1]()
  %self.conv.bias : Float(2, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=0.01 *  7.4246 -8.4862 [ CUDAFloatType{2} ]]()
  %self.conv.weight : Float(2, 64, 1, 1, strides=[64, 1, 1, 1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.conv.weight : Float(32, 3, 3, 3, strides=[27, 9, 3, 1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.conv.bias.15 : NoneType = prim::Constant()
  %self.block1.norm.running_var : Float(32, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.norm.running_mean : Float(32, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block1.norm.training.15 : bool = prim::Constant[value=0]()
  %452 : float = prim::Constant[value=0.10000000000000001]()
  %453 : float = prim::Constant[value=1.0000000000000001e-05]()
  %455 : int[] = prim::Constant[value=[2, 2]]()
  %self.block2.conv.weight : Float(64, 32, 3, 3, strides=[288, 9, 3, 1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block2.norm.running_var : Float(64, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %self.block2.norm.running_mean : Float(64, strides=[1], requires_grad=0, device=cuda:0) = prim::Constant[value=<Tensor>]()
  %460 : str[] = prim::Constant[value=["nearest", "area", "nearest-exact"]]()
  %461 : str = prim::Constant[value="bilinear"]()
  %462 : str = prim::Constant[value="builtins.ValueError"]()
  %463 : str = prim::Constant[value="align_corners option can only be set with the interpolating modes: linear | bilinear | bicubic | trilinear"]()
  %464 : int = prim::Constant[value=2]()
  %579 : bool = prim::Constant[value=0]()
  %580 : int[] = prim::Constant[value=[0, 0]]()
  %581 : Tensor = aten::_convolution(%x.1, %self.block1.conv.weight, %self.block1.conv.bias.15, %16, %16, %16, %579, %580, %17, %579, %579, %579, %579)
  %out1.2 : Tensor = aten::batch_norm(%581, %self.block1.norm.running_var, %self.block1.norm.running_mean, %self.block1.norm.running_mean, %self.block1.norm.running_var, %self.block1.norm.training.15, %452, %453, %18) 
  %467 : Tensor = aten::relu(%out1.2)
  %out.2 : Tensor = aten::max_pool2d(%467, %455, %455, %14, %16, %self.block1.norm.training.15)
  %582 : bool = prim::Constant[value=0]()
  %583 : int[] = prim::Constant[value=[0, 0]]()
  %584 : Tensor = aten::_convolution(%out.2, %self.block2.conv.weight, %self.block1.conv.bias.15, %16, %16, %16, %582, %583, %17, %582, %582, %582, %582)
  %out0.1 : Tensor = aten::batch_norm(%584, %self.block2.norm.running_var, %self.block2.norm.running_mean, %self.block2.norm.running_mean, %self.block2.norm.running_var, %self.block1.norm.training.15, %452, %453, %18)
  %471 : Tensor = aten::relu(%out0.1)
  %out0.2 : Tensor = aten::max_pool2d(%471, %455, %455, %14, %16, %self.block1.norm.training.15)
  %474 : bool? = prim::Uninitialized() # :0:0
  %475 : bool = prim::Uninitialized() # :0:0
  %476 : bool = aten::__contains__(%460, %461)
  %align_corners0.1 : bool? = prim::If(%476)
    block0():
       = prim::RaiseException(%463, %462)
      -> (%474)
    block1():
      -> (%self.block1.norm.training.15)
  %478 : int = aten::dim(%out0.2)
  %dim.2 : int = aten::sub(%478, %464)
  %scale_factors2.2 : float[] = prim::ListConstruct()
   = prim::Loop(%dim.2, %18)
    block0(%49 : int):
      %50 : float[] = aten::append(%scale_factors2.2, %4)
      -> (%18)
  %480 : int = prim::Constant[value=4]()
  %482 : str = prim::Constant[value="Input Error: Only 3D, 4D and 5D input Tensors supported (got {}D) for the modes: nearest | linear | bilinear | bicubic | trilinear | area | nearest-exact (got {})"]()
  %483 : int = prim::Constant[value=5]()
  %485 : str = prim::Constant[value="AssertionError: "]()
  %486 : str = prim::Constant[value="builtins.NotImplementedError"]()
  %487 : str = prim::Constant[value="Got 3D input, but bilinear mode needs 4D input"]()
  %488 : int = prim::Constant[value=3]()
  %489 : str = prim::Constant[value="Got 5D input, but bilinear mode needs 4D input"]()
  %494 : bool = aten::eq(%478, %480)
  %571 : bool = aten::__isnot__(%align_corners0.1, %self.block1.conv.bias.15)
  %align_corners6.2 : bool = prim::If(%571)
    block0():
      %align_corners7.3 : bool = prim::unchecked_cast(%align_corners0.1)
      -> (%align_corners7.3)
    block1():
       = prim::RaiseException(%485, %self.block1.conv.bias.15)
      -> (%475)
  %574 : Tensor = aten::upsample_bilinear2d(%out0.2, %self.block1.conv.bias.15, %align_corners6.2, %scale_factors2.2) 
   = prim::If(%494)
    block0():
       = prim::If(%571)
        block0():
          -> ()
        block1():
           = prim::RaiseException(%485, %self.block1.conv.bias.15)
          -> ()
      -> ()
    block1():
      %500 : bool = aten::eq(%478, %488)
       = prim::If(%500)
        block0():
           = prim::RaiseException(%487, %486)
          -> ()
        block1():
          -> ()
      %501 : bool = aten::eq(%478, %483)
       = prim::If(%501)
        block0():
           = prim::RaiseException(%489, %486)
          -> ()
        block1():
          -> ()
      %502 : str = aten::format(%482, %478, %461)
       = prim::RaiseException(%502, %486)
      -> ()
  %align_corners0 : bool? = prim::If(%self.block1.norm.training.15)
    block0():
       = prim::RaiseException(%463, %462)
      -> (%474)
    block1():
      -> (%self.block1.norm.training.15)
  %504 : int = aten::dim(%574)
  %dim.1 : int = aten::sub(%504, %464)
  %scale_factors2.1 : float[] = prim::ListConstruct()
   = prim::Loop(%dim.1, %18)
    block0(%64 : int):
      %65 : float[] = aten::append(%scale_factors2.1, %4)
      -> (%18)
  %516 : bool = aten::eq(%504, %480)
  %575 : bool = aten::__isnot__(%align_corners0, %self.block1.conv.bias.15)
  %align_corners6.6 : bool = prim::If(%575)
    block0():
      %align_corners7.7 : bool = prim::unchecked_cast(%align_corners0)
      -> (%align_corners7.7)
    block1():
       = prim::RaiseException(%485, %self.block1.conv.bias.15)
      -> (%475)
  %578 : Tensor = aten::upsample_bilinear2d(%574, %self.block1.conv.bias.15, %align_corners6.6, %scale_factors2.1)
   = prim::If(%516)
    block0():
       = prim::If(%575)
        block0():
          -> ()
        block1():
           = prim::RaiseException(%485, %self.block1.conv.bias.15)
          -> ()
      -> ()
    block1():
      %522 : bool = aten::eq(%504, %488)
       = prim::If(%522)
        block0():
           = prim::RaiseException(%487, %486)
          -> ()
        block1():
          -> ()
      %523 : bool = aten::eq(%504, %483)
       = prim::If(%523)
        block0():
           = prim::RaiseException(%489, %486)
          -> ()
        block1():
          -> ()
      %524 : str = aten::format(%482, %504, %461)
       = prim::RaiseException(%524, %486)
      -> ()
  %585 : bool = prim::Constant[value=0]()
  %586 : int[] = prim::Constant[value=[0, 0]]()
  %587 : Tensor = aten::_convolution(%578, %self.conv.weight, %self.conv.bias, %16, %14, %16, %585, %586, %17, %585, %585, %585, %585)
  return (%587)
```
</details>

--- a/core/lowering/lowering.cpp
+++ b/core/lowering/lowering.cpp
@@ -9,7 +9,6 @@
 #include "torch/csrc/jit/passes/lower_graph.h"
 #include "torch/csrc/jit/passes/lower_tuples.h"
 #include "torch/csrc/jit/passes/peephole.h"
-#include "torch/csrc/jit/passes/remove_exceptions.h"
 #include "torch/csrc/jit/passes/remove_mutation.h"
 
 #include "core/lowering/lowering.h"
@@ -105,7 +104,7 @@ void LowerGraph(std::shared_ptr<torch::jit::Graph>& g, std::vector<torch::jit::I
   torch::jit::InlineFunctionalGraphs(g);
   torch::jit::PeepholeOptimize(g, false);
   torch::jit::FuseLinear(g);
-  torch::jit::EliminateExceptions(g);
+  passes::EliminateExceptionsSafe(g);
   if (!lower_info.disable_cse) {
     torch::jit::EliminateCommonSubexpression(g);
   }

--- a/core/lowering/passes/exception_elimination.cpp
+++ b/core/lowering/passes/exception_elimination.cpp
@@ -1,5 +1,6 @@
 #include "torch/csrc/jit/ir/alias_analysis.h"
 #include "torch/csrc/jit/jit_log.h"
+#include "torch/csrc/jit/passes/constant_pooling.h"
 #include "torch/csrc/jit/passes/constant_propagation.h"
 #include "torch/csrc/jit/passes/dead_code_elimination.h"
 #include "torch/csrc/jit/passes/guard_elimination.h"
@@ -106,6 +107,71 @@ void EliminateExceptionOrPassPattern(std::shared_ptr<Graph> graph) {
   if (graph) {
     LOG_GRAPH("Post Eliminate Exception or Pass Patterns: " << *graph);
   }
+}
+
+/*
+  Below is a fork of the torch::jit::EliminateExceptions pass, with node replacement
+  using replaceAllUsesDominatedByNodeWith instead of replaceAllUsesWith,
+  so as to not invalidate the IR in challenging cases, such as nested Ifs
+
+  Original Source from which it was adapted:
+  https://github.com/pytorch/pytorch/blob/c29ab84115f40614d04e4557ea2e1ac40b7aa75c/torch/csrc/jit/passes/remove_exceptions.cpp
+*/
+
+bool certainlyThrows(Block* block) {
+  // A block certainly throws an exception if it contains
+  // the prim::RaiseException operation
+  for (Node* n : block->nodes()) {
+    if (n->kind() == prim::RaiseException) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void EliminateExceptionsSafe(Block* block) {
+  auto graph = block->owningGraph();
+  // Generate false and true constant placeholders
+  Value* false_const = graph->insertConstant(IValue(false));
+  Value* true_const = graph->insertConstant(IValue(true));
+
+  // For each prim::If node, if either block certainly throws an exception,
+  // replace input conditional of the node input with the logical opposite
+  for (Node* n : block->nodes()) {
+    if (n->kind() == prim::If) {
+      Block* true_block = n->blocks()[0];
+      Block* false_block = n->blocks()[1];
+      bool removed_exception = false;
+      Value* input_value_replacement;
+
+      // If the block throws an exception, replace input with logical opposite
+      if (certainlyThrows(true_block)) {
+        removed_exception = true;
+        input_value_replacement = false_const;
+      } else if (certainlyThrows(false_block)) {
+        removed_exception = true;
+        input_value_replacement = true_const;
+      }
+
+      // Log node and perform input replacement
+      if (removed_exception) {
+        LOG_WARNING("Detected and removing exception in TorchScript IR for node: " << util::node_info(n));
+        n->insertInput(0, input_value_replacement);
+        n->removeInput(1);
+      }
+    }
+
+    // Inspect and replace all instances within subblocks of the current node
+    for (Block* subblock : n->blocks()) {
+      EliminateExceptionsSafe(subblock);
+    }
+  }
+}
+
+void EliminateExceptionsSafe(std::shared_ptr<Graph>& graph) {
+  EliminateExceptionsSafe(graph->block());
+  ConstantPropagation(graph);
+  ConstantPooling(graph);
 }
 
 } // namespace passes

--- a/core/lowering/passes/passes.h
+++ b/core/lowering/passes/passes.h
@@ -20,6 +20,7 @@ void Conv3DToConvolution(std::shared_ptr<torch::jit::Graph>& graph);
 void ConvTransposed3DToConvolution(std::shared_ptr<torch::jit::Graph>& graph);
 void FuseAddMMBranches(std::shared_ptr<torch::jit::Graph> graph);
 void LinearToAddMM(std::shared_ptr<torch::jit::Graph>& graph);
+void EliminateExceptionsSafe(std::shared_ptr<torch::jit::Graph>& graph);
 void EliminateExceptionOrPassPattern(std::shared_ptr<torch::jit::Graph> graph);
 void ReduceToOperation(std::shared_ptr<torch::jit::Graph>& graph);
 void ReduceGelu(std::shared_ptr<torch::jit::Graph>& graph);

--- a/tests/core/lowering/test_exception_elimination_pass.cpp
+++ b/tests/core/lowering/test_exception_elimination_pass.cpp
@@ -1,6 +1,10 @@
 #include "core/lowering/passes/passes.h"
 #include "gtest/gtest.h"
+#include "tests/util/util.h"
 #include "torch/csrc/jit/ir/irparser.h"
+#include "torch/csrc/jit/passes/canonicalize.h"
+#include "torch/csrc/jit/passes/constant_pooling.h"
+#include "torch/csrc/jit/passes/remove_exceptions.h"
 
 TEST(LoweringPasses, EliminateExceptionOrPassPattern_Block0) {
   // parseIR does not support " = prim::If(%51)" with no return value
@@ -168,4 +172,218 @@ TEST(LoweringPasses, EliminateExceptionOrPassPattern_Negative) {
     }
   }
   EXPECT_EQ(1, if_count);
+}
+
+TEST(LoweringPasses, EliminateExceptionsSafeIfBlock) {
+  /*std::string source_graph = R"IR(
+    graph(%x, %y):
+      %dim : int = aten::dim(%x)
+      %48 : int = prim::Constant[value=2]()
+      %66 : bool = aten::eq(%48, %dim)
+      %45 : str = prim::Constant[value="EXCEPTION"]()
+      %4 : Tensor = prim::If(%66)
+        block0():
+          = prim::RaiseException(%45)
+          -> (%x)
+        block1():
+          %res = aten::mul(%x, %y)
+          -> (%res)
+      return (%4))IR";*/
+
+  std::string target_graph = R"IR(
+    graph(%x : Tensor,
+          %y : Tensor):
+      %6 : Tensor = aten::mul(%x, %y)
+      return (%6))IR";
+
+  // Construct graph via manual commands, to avoid IR parsing issues with
+  // unassigned variables (such as prim::RaiseException)
+  auto g = std::make_shared<torch::jit::Graph>();
+  auto x = g->insertInput(0, "x");
+  auto y = g->insertInput(1, "y");
+  auto none_const_val = g->insertConstant(torch::jit::IValue());
+  auto two_const_val = g->insertConstant(torch::jit::IValue(2));
+  auto x_dims = g->create(torch::jit::aten::dim, {x}, 1);
+  g->appendNode(x_dims);
+  x_dims->output()->setType(torch::jit::IntType::get());
+  auto eq = g->create(torch::jit::aten::eq, {two_const_val, x_dims->output()}, 1);
+  g->appendNode(eq);
+  eq->output()->setType(torch::jit::BoolType::get());
+  torch::jit::IValue except("EXCEPTION");
+  auto except_val = g->insertConstant(except);
+
+  auto if_node = g->create(torch::jit::prim::If, {eq->output()}, 1);
+  auto if_block0 = if_node->addBlock();
+  auto exception_node = g->create(torch::jit::prim::RaiseException, {except_val, none_const_val}, 0);
+  if_block0->appendNode(exception_node);
+  if_block0->registerOutput(x);
+
+  auto if_block1 = if_node->addBlock();
+  auto sum_node = g->create(torch::jit::aten::mul, {x, y}, 1);
+  if_block1->appendNode(sum_node);
+  if_block1->registerOutput(sum_node->output());
+
+  g->insertNode(if_node);
+  g->registerOutput(if_node->output());
+
+  // Apply lowering pass and canonicalization to the graph
+  torch_tensorrt::core::lowering::passes::EliminateExceptionsSafe(g);
+  g = torch::jit::Canonicalize(g, false);
+
+  auto tg = std::make_shared<torch::jit::Graph>();
+  torch::jit::parseIR(target_graph, tg.get());
+
+  torch::jit::ConstantPooling(tg);
+  tg = torch::jit::Canonicalize(tg, false);
+
+  // Validate identical graphs after pooling constants and canonicalizing
+  ASSERT_TRUE((tg->toString() == g->toString()));
+}
+
+TEST(LoweringPasses, EliminateExceptionsSafeElseBlock) {
+  /*std::string source_graph = R"IR(
+    graph(%x, %y):
+      %dim : int = aten::dim(%x)
+      %48 : int = prim::Constant[value=2]()
+      %66 : bool = aten::eq(%48, %dim)
+      %45 : str = prim::Constant[value="EXCEPTION"]()
+      %4 : Tensor = prim::If(%66)
+        block0():
+          %res = aten::matmul(%x, %y)
+          -> (%res)
+        block1():
+          = prim::RaiseException(%45)
+          -> (%x)
+      return (%4))IR";*/
+
+  std::string target_graph = R"IR(
+    graph(%x : Tensor,
+          %y : Tensor):
+      %6 : Tensor = aten::matmul(%x, %y)
+      return (%6))IR";
+
+  // Construct graph via manual commands, to avoid IR parsing issues with
+  // unassigned variables (such as prim::RaiseException)
+  auto g = std::make_shared<torch::jit::Graph>();
+  auto x = g->insertInput(0, "x");
+  auto y = g->insertInput(1, "y");
+  auto none_const_val = g->insertConstant(torch::jit::IValue());
+  auto two_const_val = g->insertConstant(torch::jit::IValue(2));
+  auto x_dims = g->create(torch::jit::aten::dim, {x}, 1);
+  g->appendNode(x_dims);
+  x_dims->output()->setType(torch::jit::IntType::get());
+  auto eq = g->create(torch::jit::aten::eq, {two_const_val, x_dims->output()}, 1);
+  g->appendNode(eq);
+  eq->output()->setType(torch::jit::BoolType::get());
+  torch::jit::IValue except("EXCEPTION");
+  auto except_val = g->insertConstant(except);
+
+  auto if_node = g->create(torch::jit::prim::If, {eq->output()}, 1);
+  auto if_block0 = if_node->addBlock();
+  auto sum_node = g->create(torch::jit::aten::matmul, {x, y}, 1);
+  if_block0->appendNode(sum_node);
+  if_block0->registerOutput(sum_node->output());
+
+  auto if_block1 = if_node->addBlock();
+  auto exception_node = g->create(torch::jit::prim::RaiseException, {except_val, none_const_val}, 0);
+  if_block1->appendNode(exception_node);
+  if_block1->registerOutput(x);
+
+  g->insertNode(if_node);
+  g->registerOutput(if_node->output());
+
+  // Apply lowering pass and canonicalization to the graph
+  torch_tensorrt::core::lowering::passes::EliminateExceptionsSafe(g);
+  g = torch::jit::Canonicalize(g, false);
+
+  auto tg = std::make_shared<torch::jit::Graph>();
+  torch::jit::parseIR(target_graph, tg.get());
+
+  torch::jit::ConstantPooling(tg);
+  tg = torch::jit::Canonicalize(tg, false);
+
+  // Validate identical graphs after pooling constants and canonicalizing
+  ASSERT_TRUE((tg->toString() == g->toString()));
+}
+
+TEST(LoweringPasses, EliminateExceptionsSafeNestedIfBlock) {
+  /*std::string source_graph = R"IR(
+    graph(%x, %y):
+      %false : bool = prim::Constant[value=0]()
+      %dim : int = aten::dim(%x)
+      %48 : int = prim::Constant[value=2]()
+      %66 : bool = aten::eq(%48, %dim)
+      %45 : str = prim::Constant[value="EXCEPTION"]()
+      %4 : Tensor = prim::If(%66)
+        block0():
+          %45 : str = prim::Constant[value="EXCEPTION"]()
+          = prim::If(%false)
+            block0():
+              -> ()
+            block1():
+              = prim::RaiseException(%45)
+              -> ()
+          = prim::RaiseException(%45)
+          -> (%x)
+        block1():
+          %res = aten::mul(%x, %y)
+          -> (%res)
+      return (%4))IR";*/
+
+  std::string target_graph = R"IR(
+    graph(%x : Tensor,
+          %y : Tensor):
+      %6 : Tensor = aten::mul(%x, %y)
+      return (%6))IR";
+
+  // Construct graph via manual commands, to avoid IR parsing issues with
+  // unassigned variables (such as prim::RaiseException)
+  auto g = std::make_shared<torch::jit::Graph>();
+  auto x = g->insertInput(0, "x");
+  auto y = g->insertInput(1, "y");
+  auto none_const_val = g->insertConstant(torch::jit::IValue());
+  auto false_const_val = g->insertConstant(torch::jit::IValue(false));
+  auto two_const_val = g->insertConstant(torch::jit::IValue(2));
+  auto x_dims = g->create(torch::jit::aten::dim, {x}, 1);
+  g->appendNode(x_dims);
+  x_dims->output()->setType(torch::jit::IntType::get());
+  auto eq = g->create(torch::jit::aten::eq, {two_const_val, x_dims->output()}, 1);
+  g->appendNode(eq);
+  eq->output()->setType(torch::jit::BoolType::get());
+  torch::jit::IValue except("EXCEPTION");
+  auto except_val = g->insertConstant(except);
+
+  // Construct nested-If substructure in graph
+  auto if_node = g->create(torch::jit::prim::If, {eq->output()}, 1);
+  auto if_block0 = if_node->addBlock();
+  auto if_if_node = g->create(torch::jit::prim::If, {false_const_val}, 0);
+  if_block0->appendNode(if_if_node);
+  /* auto if_if_block0 = */ if_if_node->addBlock();
+  auto if_if_block1 = if_if_node->addBlock();
+  auto exception_node = g->create(torch::jit::prim::RaiseException, {except_val, none_const_val}, 0);
+  if_if_block1->appendNode(exception_node);
+  auto exception_node_2 = g->create(torch::jit::prim::RaiseException, {except_val, none_const_val}, 0);
+  if_block0->appendNode(exception_node_2);
+  if_block0->registerOutput(x);
+
+  auto if_block1 = if_node->addBlock();
+  auto sum_node = g->create(torch::jit::aten::mul, {x, y}, 1);
+  if_block1->appendNode(sum_node);
+  if_block1->registerOutput(sum_node->output());
+
+  g->insertNode(if_node);
+  g->registerOutput(if_node->output());
+
+  // Apply lowering pass and canonicalization to the graph
+  torch_tensorrt::core::lowering::passes::EliminateExceptionsSafe(g);
+  g = torch::jit::Canonicalize(g, false);
+
+  auto tg = std::make_shared<torch::jit::Graph>();
+  torch::jit::parseIR(target_graph, tg.get());
+
+  torch::jit::ConstantPooling(tg);
+  tg = torch::jit::Canonicalize(tg, false);
+
+  // Validate identical graphs after pooling constants and canonicalizing
+  ASSERT_TRUE((tg->toString() == g->toString()));
 }

